### ISSUE TITLE
amp-video insists on local videos being served from localhost

### ISSUE THREAD

**Issue by dandv** [ampproject/amphtml#20543]
I'm spinning up a local server for static files, and it's listening to http://127.0.0.1:8080.

When I navigate to a page containing an `<amp-video src="local/path/video.mp4>`, I see this error in the console:

> "source" "must start with "https://" or "//" or be relative and served from either https or from localhost. Invalid value:" "assets/template-howto/videos/dummy-video.mp4"

![image](https://user-images.githubusercontent.com/33569/51781218-97933380-20ca-11e9-92da-ddc23396cdb4.png)

If I navigate to http://localhost:8080, the errors are not shown.

**Comment by jpettitt** [comment]
I think [this](https://github.com/ampproject/amphtml/blob/e2e6ab072076477a522707fdb212b18c67ead046/src/url.js#L301) is the offending code.  It looks for `^//`, `https:` or `localhost`.  A relative url will fail that assert.

```JAVASCRIPT
userAssert(isSecureUrlDeprecated(theUrlString) || /^(\/\/)/.test(theUrlString),
```

**Comment by jridgewell** [comment]
We should add it to `isSecureUrlDeprecated`. This is a good first issue, if you don't mind waiting @dandv?

**Comment by moose-codes** [comment]
Heya! I would love to start contributing to this project, could I start with this issue?

**Comment by mrjoro** [comment]
@moose-codes I'd say go for it!  Ping @jridgewell if you have any questions.

**Comment by jridgewell** [comment]
Closed by #21757

**Comment by D3MZ** [comment]
Python3 default local server address is now (http://0.0.0.0:8000/), could you add an exception for that as well too please?

Anything that's not binding to port 80 might be an okay catch-all solution?
